packages/astro/src/runtime.ts

```diff
@@ -99,6 +99,23 @@ async function load(config: RuntimeConfig, rawPathname: string | undefined): Pro
     let collection = {} as CollectionResult;
     let additionalURLs = new Set<string>();
 
+    if (mod.exports.__content) {
+      const hasDraftAttribute = mod.exports.__content.hasOwnProperty('draft');
+      const hasPublishAttribute = mod.exports.__content.hasOwnProperty('published');
+      if(hasPublishAttribute && mod.exports.__content['published'] === false) {
+        return { 
+          statusCode: 404, 
+          error: new Error('Unpublished document')
+        };
+      }
+      if(hasDraftAttribute && mod.exports.__content['draft'] === true && buildOptions.draft && config.mode === 'production') {
```

Comment:
Won't this throw whenever you try to build a site with docs in draft mode? I think the way people will work is they will have documents that are in draft, possibly for quite a while, and we should simple not build those. 

It might be better to let the build and dev handle what they want to do with drafts. Possibly by adding a `draft: true` to the return object. The build could use that info to decide to ignore the page. The dev server likely still renders it.

Reply:
I see ... currently the build ignores the page by identifying the error message of a 404 being 'Unpublished document' which I don't like that much. But it works ... 😆 
I wanted to use the type: 'document-unpublished' as part of the `LoadResult` object for that however interpreting the type attribute did not work properly in the `page.ts`. I can invest this again.